### PR TITLE
Cherry-pick fix for wrong narration when legend selected

### PR DESCRIPTION
**Pull request by krkshitij**
Cherry-pick of #24903 and #25978

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">7.0 Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/BaseButton-mount_base.html">925</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/BaseButton-mount.html">911</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Breadcrumb-mount_base.html">45072</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Breadcrumb-mount.html">44484</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Checkbox-mount_base.html">1617</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Checkbox-mount.html">1645</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/CheckboxBase-mount_base.html">1321</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/CheckboxBase-mount.html">1309</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/ChoiceGroup-mount_base.html">5090</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/ChoiceGroup-mount.html">5232</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/ComboBox-mount_base.html">952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/ComboBox-mount.html">919</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/CommandBar-mount_base.html">8364</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/CommandBar-mount.html">8609</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/ContextualMenu-mount_base.html">17846</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/ContextualMenu-mount.html">17717</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DefaultButton-mount_base.html">1133</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DefaultButton-mount.html">1148</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DetailsRow-mount_base.html">3918</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DetailsRow-mount.html">3871</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DetailsRowFast-mount_base.html">3988</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DetailsRowFast-mount.html">3883</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DetailsRowNoStyles-mount_base.html">3716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DetailsRowNoStyles-mount.html">3771</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Dialog-mount_base.html">2353</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Dialog-mount.html">2402</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DocumentCardTitle-mount_base.html">2036</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/DocumentCardTitle-mount.html">2026</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Dropdown-mount_base.html">2572</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Dropdown-mount.html">2741</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/FocusTrapZone-mount_base.html">1821</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/FocusTrapZone-mount.html">1818</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/FocusZone-mount_base.html">1918</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/FocusZone-mount.html">1950</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/IconButton-mount_base.html">1810</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/IconButton-mount.html">1851</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Label-mount_base.html">344</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Label-mount.html">348</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Layer-mount_base.html">2060</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Layer-mount.html">2078</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Link-mount_base.html">447</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Link-mount.html">448</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/MenuButton-mount_base.html">1445</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/MenuButton-mount.html">1543</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/MessageBar-mount_base.html">2203</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/MessageBar-mount.html">2147</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Nav-mount_base.html">3340</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Nav-mount.html">3343</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/OverflowSet-mount_base.html">1487</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/OverflowSet-mount.html">1468</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Panel-mount_base.html">1556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Panel-mount.html">1530</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Persona-mount_base.html">983</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Persona-mount.html">995</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Pivot-mount_base.html">1489</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Pivot-mount.html">1463</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/PrimaryButton-mount_base.html">1334</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/PrimaryButton-mount.html">1321</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Rating-mount_base.html">7909</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Rating-mount.html">7906</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SearchBox-mount_base.html">1293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SearchBox-mount.html">1279</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Shimmer-mount_base.html">2616</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Shimmer-mount.html">2643</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Slider-mount_base.html">1532</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Slider-mount.html">1524</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SpinButton-mount_base.html">5163</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SpinButton-mount.html">5073</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Spinner-mount_base.html">416</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Spinner-mount.html">420</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SplitButton-mount_base.html">3212</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SplitButton-mount.html">3256</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Stack-mount_base.html">503</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Stack-mount.html">500</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/StackWithIntrinsicChildren-mount_base.html">1698</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/StackWithIntrinsicChildren-mount.html">1676</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/StackWithTextChildren-mount_base.html">4784</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/StackWithTextChildren-mount.html">4777</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SwatchColorPicker-mount_base.html">10593</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/SwatchColorPicker-mount.html">10591</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/TagPicker-mount_base.html">2854</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/TagPicker-mount.html">2918</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/TeachingBubble-mount_base.html">51798</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/TeachingBubble-mount.html">51887</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Text-mount_base.html">415</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Text-mount.html">423</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/TextField-mount_base.html">1356</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/TextField-mount.html">1390</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Toggle-mount_base.html">817</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/Toggle-mount.html">827</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/button-mount_base.html">117</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25936/perf-test/button-mount.html">120</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: change/@uifabric-charting-1027db92-a05d-490c-bb5c-874afb13b7d9.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Fix wrong narration when legend selected",
+  "packageName": "@uifabric/charting",
```
Comment (AtishayMsft) [inline change/@uifabric-charting-1027db92-a05d-490c-bb5c-874afb13b7d9.json:4]: From what I remember, doesn't this change depend on the change where you added active/highlighted legend states.

Comment (krkshitij) [reply change/@uifabric-charting-1027db92-a05d-490c-bb5c-874afb13b7d9.json:4]: Actually, it was the other way around. #24563 depended on this change

**Comment by fabricteam** [comment]
## 🕵 FluentUI-v7 No visual regressions between this PR and main
<div id="vrtCommentFluentUI-v7"/>

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**